TempleEight/docs/arch/system.md

```diff
@@ -3,3 +3,190 @@ id: system
+    baz: int;
+}
+```
+
+This has the effect of generating inter-service communication logic in the service-level application code.
+On the project level, nothing changes, except the services use the hostname resolution of the orchestration platform to make HTTP requests to each other.
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-foreign-key.png')} />
+
+### Structs
+
+Adding a struct to a service in your Templefile represents two data items that are strongly coupled (see the [Struct](../guide/structs) guide for details).
+This is implemented in Temple generated projects by extending the parent's application executable to include the additional endpoints required to process struct requests.
+The struct's data is stored in a separate table of the service's database. 
+This means that when adding structs to your Templefile doesn't change the overall system-level architecture at all.
+
+### Auth
+
+As per the [Authentication](../guide/authentication) Guide, 
+authentication can be added to your system by augmenting your Templefile to include `#auth` metadata on services, and a `#authMethod` annotation on the project block.
+For example:
+
+```templefile {5}{10}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+  #authMethod(email);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+  #auth;
+}
+```
+
+These additions result in an additional `Auth` service being generated automatically that stores and verifies user credentials, and issues authentication tokens to users for use with other services.
+This is implemented by configuring Kong to check that any incoming requests have a valid [JWT](https://jwt.io/), except those destined for the auth service. 
+Any requests that are not correctly authenticated do not pass Kong, so the internal microservices can be sure that all traffic has been correctly authenticated.
+
+Visually, the changes to our architecture look like:
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-auth.png')} />
+
+### Metrics
+
+Temple also includes the ability to generate metrics monitoring suites, using [Prometheus](https://prometheus.io/) and [Grafana](https://grafana.com/).
+For full details of their use and implementations, check out the [Metrics](../guide/metrics) guide.
+
+Let's add metrics to our example and see how the output changes.
+
+```templefile {5}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+  #metrics(prometheus);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+When a project has metrics, two more infrastructure components are needed, a Prometheus instance and a Grafana instance. 
+Prometheus is a metrics aggregator.
+Each service contains logic to aggregate metrics about it's performance, and forwards them on to the Prometheus instance, which collects and stores the metrics from all the services.
```

Comment:
Hm be careful, Prometheus is a pull system not a push system. I wouldn't say "forwards on"

Reply:
Maybe something like makes them accessible to prometheus